core/src/main/java/org/infinispan/util/concurrent/locks/impl/DefaultLockManager.java

```diff
@@ -346,6 +374,21 @@ public void addListener(LockListener listener) {
       }
 
       @Override
+      public InvocationStage toInvocationStage() {
+         if (notifier.isDone()) {
+            return checkState(notifier.getNow(lockState), SyncInvocationStage::new, SimpleAsyncInvocationStage::new);
+         } else {
+            return new SimpleAsyncInvocationStage(notifier.thenApplyAsync(lockState -> {
```

Comment:
`thenApplyAsync` should be `thenApply`

Reply:
no! it would create deadlocks! the notified is completed inside a CHM.compute()

Reply:
I agree that we shouldn't use FJM pool, providing a remote executor here would be better.